core/orchestrator.go

```diff
@@ -102,26 +102,43 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 		return nil
 	}
 
-	ticket := &pm.Ticket{
-		Recipient:         ethcommon.BytesToAddress(payment.Ticket.Recipient),
-		Sender:            ethcommon.BytesToAddress(payment.Ticket.Sender),
-		FaceValue:         new(big.Int).SetBytes(payment.Ticket.FaceValue),
-		WinProb:           new(big.Int).SetBytes(payment.Ticket.WinProb),
-		SenderNonce:       payment.Ticket.SenderNonce,
-		RecipientRandHash: ethcommon.BytesToHash(payment.Ticket.RecipientRandHash),
+	if payment.TicketParams == nil {
+		return errors.Wrapf(nil, "Payment TicketParams is nil")
 	}
-	seed := new(big.Int).SetBytes(payment.Seed)
 
-	sessionID, won, err := orch.node.Recipient.ReceiveTicket(ticket, payment.Sig, seed)
-	if err != nil {
-		return errors.Wrapf(err, "error receiving ticket for payment %v for manifest %v", payment, manifestID)
+	if payment.Sender == nil {
+		return errors.Wrapf(nil, "Sender address is nil")
 	}
 
-	if won {
-		glog.V(common.DEBUG).Info("Received winning ticket")
-		cachePMSessionID(orch.node, manifestID, sessionID)
+	seed := new(big.Int).SetBytes(payment.TicketParams.Seed)
+
+	ticket := &pm.Ticket{
+		Recipient:         ethcommon.BytesToAddress(payment.TicketParams.Recipient),
+		Sender:            ethcommon.BytesToAddress(payment.Sender),
+		FaceValue:         new(big.Int).SetBytes(payment.TicketParams.FaceValue),
+		WinProb:           new(big.Int).SetBytes(payment.TicketParams.WinProb),
+		SenderNonce:       0,
+		RecipientRandHash: ethcommon.BytesToHash(payment.TicketParams.RecipientRandHash),
 	}
 
+	for _, tsp := range payment.TicketSenderParams {
```

Comment:
Test for a payment with multiple tickets? We could add a test in `orch_test.go` that checks if `orch.node.Recipient.ReceiveTicket()` is called N times if there are N `TicketSenderParams` in the payment

Reply:
Added according to the docs you sent me.